### PR TITLE
(#2776, #2834)  Title first feature card row and spacing fixes

### LINKED ISSUE #2776: Story: Title-First Feature Card Row
## As a content editor, I would like to be able to embed a title-first feature card row in content, so that we can promote content more prominently.  

> **ESTIMATE** 8

### Details

- New paragraph option
- Two slots - 2 column card, then one column card
- Only allows internal and external feature cards
- Back end fields remain the same - allows for all overrides
- Display modeled after media cards on current home and landing, but different mobile presentation
  - desktop -  a 2 column card and a 1 column card
  - mobile - stacks to single-column cards
- Both cards pull 16:9 image crop
- Whole card is clickable - title color change on hover

### Acceptance Criteria

```gherkin
Scenario: User views a page with a title-first feature card row on desktop or tablet 
  Given a user is viewing the AP landing page 
    And the browser width is 768px or greater 
  Then the title-first feature card row will display 
    And a two-column card will appear on the left column of the main content area 
    And the card title of the selected feature card item should appear as a H3 (following our card title rules) 
    And the 16:9 crop of the promo image of the selected feature card item should appear (following our image rules) 
    And the card description of the selected feature card item should appear 
    And the entire card will link to the selected feature card item 
    And the card title will turn cranberry on hover or focus 
  Then the one-column card will appear in the right column of the main content area 
    And the card title of the selected feature card item should appear as a H3 (following our card title rules) 
    And the 16:9 crop off the promo image of the selected feature card item should appear (following our image rules) 
    And the card description of the selected feature card item should appear 
    And the entire card will link to the selected feature card item 
    And the card title will turn cranberry on hover or focus 
```

```gherkin
Scenario: User views a page with title-first feature card row on mobile 
  Given a user is viewing the AP landing page 
    And the browser width is 640px or less 
  Then the title-first feature card row will stack 
    And the two-column card will appear 
    And the one-column card will appear 
```

```gherkin
Scenario: Internal feature card, no overrides 
  Given a user is viewing a page with a Title-First Feature Card Row 
    And one of the cards is an Internal Feature Card 
    And the content editor has not put content in any of the override fields 
  Then the featured item’s Card Title will appear 
    And the 16:9 crop of the promo image of the selected feature card item will appear (following our image rules) 
    And the card description of the selected feature card item will appear 
```

```gherkin
Scenario: Internal feature card, overrides 
  Given a user is viewing a page with a Title-First Feature Card Row 
    And one of the cards is an Internal Feature Card 
    And the content editor has filled in the override fields 
  Then the featured item’s Override Card Title will appear 
    And the 16:9 crop of the Override Promotional Image will appear (following our image rules) 
    And the Override Card Description will appear 
```

```gherkin
Scenario: External feature card 
  Given a user is viewing a page with a Title-First Feature Card Row 
    And one of the cards is an External Feature Card 
  Then the content in the card's Card Title field will appear 
    And the 16:9 crop of the promo image associated with the card will appear 
    And the content in the card's Card Description will appear 
    And the whole card will link to the destination in the card's Featured Item URL field 
```

```gherkin
Scenario: User views a Spanish page with a Title-First Feature Card Row 
  Given language is set to "ES” 
  Then the Spanish content of all fields will display 
```

```gherkin
Scenario: Content editor adds a Title-First Feature Card Row to the AP landing page 
  Given the user is logged in as an editor 
    And the user edits the AP landing page 
    And expands the drop down to add a paragraph to Contents 
    And selects "Add Title-First Feature Card Row" from the list 
  Then the fields for a title first feature card row appear 
    And a field called "Two Column Feature Card" appears 
    And it shows that it is empty by displaying text that reads "No Feature Card Added Yet" 
    And it has a tooltip that reads "Appears aligned left on desktop" 
    And it has a dropdown button with two options: "Add Internal Feature Card" and "Add External Feature Card" 
    And the dropdown button defaults to "Add Internal Feature Card" 
    And a field called "One Column Feature Card" appears  
    And it shows that it is empty by displaying text that reads "No Feature Card Added Yet" 
    And it has a tooltip that reads "Appears aligned right on desktop" 
    And it has a dropdown button with two options: "Add Internal Feature Card" and "Add External Feature Card" 
    And the dropdown button defaults to "Add Internal Feature Card" 
```
**Analytics**
```gherkin
Scenario: Click events for feature card row clicks
Given user is viewing the AP landing page
When user clicks on a title-first feature card with an internal link
Then page click request is sent
And the following parameters should be captured
| parameter | value                    |
      | event27   |                          |
      | prop4     | D=pev1                   |
      | prop8     | [language]               |
      | prop57    | [card title]             |
      | prop58    | [card title]             |
      | prop 59   | titlefirst:card position |
      | prop60    | D=pagename               |
      | prop61    | [previous page name]     |
      | prop67    | D=pagename               |

Scenario: Click events for feature card row clicks
Given user is viewing the AP landing page
When user clicks on a title-first feature card with an external link
Then page click request is sent
And the following parameters should be captured
Click Event:
| parameter | value                    |
      | event27   |                          |
      | prop4     | D=pev1                   |
      | prop8     | [language]               |
      | prop57    | [card title]             |
      | prop58    | [card title]             |
      | prop59   | titlefirst:card position |
      | prop60    | D=pagename               |
      | prop61    | [previous page name]     |
      | prop67    | D=pagename               |

Exit Link Click:
 | parameter | value           |
      | event1    |                 |
      | prop6     | [short title]   |
      | prop8     | [language]      |
      | prop10    | [browser title] |
      | prop25    | [exit date]     |
      | prop44    | [exit page tag] |
      | eVar2     | D=c8            |
      | evar7     | D=c7            |
      | evar44    | D=c44           |
```
---
### Resources
Mockups
- Desktop: https://publicis.invisionapp.com/share/DZWC0VL2B3E#/410081138_AP_Landing_Page_V9_-_Desktop

- Tablet 1024:  https://publicis.invisionapp.com/share/DZWC0VL2B3E#/415867266_AP_Landing_Page_V9_-_Tablet_1024

- Mobile: https://publicis.invisionapp.com/share/DZWC0VL2B3E#/411216125_AP_Landing_Page_V9_-_Mobile
---
### Notes

Content Manager
- _Add as paragraph to Contents_ 
---
### Fields

- On Home and Landing Wiki: https://github.com/NCIOCPL/cgov-digital-platform/wiki/Content-Type-Home-&-Landing-Page#title-first-feature-card-row

---
### Tasks
- [ ] Add Title-First Feature Card Row paragraph type
- [ ] Add styling (define new suggestions for existing internal and external card fields)
- [ ] Add analytics for card click (need to add ACs to story)

### LINKED ISSUE #2834: Borderless card top padding to be added back in (Mobile)
### Issue description
Borderless card had the correct top padding on mobile at the end of Sprint 1, in sprint 2 the padding was no longer there. 

### What's the expected result?
- When you create a special report type home and landing page and add a borderless card, **the top of the card should have padding** on mobile so that the title doesn't rest up against the content before it.

### What's the actual result?
- When you create a special report type home and landing page and add a borderless card, **the top of the card does not have padding** on mobile and the title rests up against the content before it.

### Additional details / screenshot
- Screenshot: 
Borderless card no padding

### PR DESCRIPTION
**Pull request by adriancofie**
ODE: http://ncigovcdode366.prod.acquia-sites.com

This is a recommit of PR #2843 

* Adds the title first feature card row and related items
* Spacing fixes for borderless card

Closes #2776 
Closes #2834

### CODE REVIEW

File: docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/templates/cards/card_macros.twig [hunk 1]
```diff
@@ -51,10 +67,10 @@
 
 {% macro thumbnailCard(title, list_description, image, url) %}
 	<div
-		class="row card-thumbnail">
+			class="row card-thumbnail">
```
Comment (arcepaul) [inline docroot/profiles/custom/cgov_site/themes/custom/cgov/cgov_common/templates/cards/card_macros.twig:70]: Nit: what's going on with this formatting?

**Comment by meglake** [comment]
Confirmed complete in this PR. Please merge into a build with all sprint stories for final testing.

File: docroot/profiles/custom/cgov_site/themes/custom/cgov/src/styles/_borderless-cards.scss [hunk 2]
```diff
@@ -5,7 +5,7 @@
 .borderless-container {
   padding: 20px 0.9375em 1.5em 0.9375em;
   @include bp(small) {
-    padding: 0 0.9375em;
+    padding: 20px 0.9375em 0 0.9375em;
```
Comment (arcepaul) [inline docroot/profiles/custom/cgov_site/themes/custom/cgov/src/styles/_borderless-cards.scss:8]: Nit: that last 0.9375em is superfluous

File: docroot/profiles/custom/cgov_site/themes/custom/cgov/src/styles/_colors.scss [hunk 3]
```diff
@@ -76,4 +76,5 @@ $modal-background: #445159;
 $very-dark-grey: #363636;   //User for AP Titles
 $button-box-shadow-color: #1b5a8c;  //Used for AP Borderless card shadow color
 $light-gray-text: #666666; //Used for AP Borderless card description text
-$alternating-image-box-phone-shadow: 0 0 9px 1px rgba(0,0,0,0.15);
\ No newline at end of file
+$special-report-box-shadow: 0 0 9px 1px rgba(0,0,0,0.15);
+$alternating-image-box-phone-shadow: $special-report-box-shadow;
```
Comment (arcepaul) [inline docroot/profiles/custom/cgov_site/themes/custom/cgov/src/styles/_colors.scss:80]: Nit: needs a newline at end of document

File: docroot/profiles/custom/cgov_site/themes/custom/cgov/src/styles/_title-first-feature-cards.scss [hunk 4]
```diff
@@ -0,0 +1,68 @@
+@import "./breakpoints";
+@import "./colors";
+
+/* Title First Feature Cards */
+.title-first-feature-card-container {
+  @include flexcontainer();
+  padding: 20px 0.9375em 1.5em 0.9375em;
```
Comment (arcepaul) [inline docroot/profiles/custom/cgov_site/themes/custom/cgov/src/styles/_title-first-feature-cards.scss:7]: all the other stylesheets are using break(), why desktop down here?

**Pull request merged**